clients/unshaded/src/main/java/tachyon/client/lineage/TachyonLineageFileSystem.java

```diff
@@ -0,0 +1,164 @@
+      return lineageId;
+    } finally {
+      mContext.releaseMasterClient(masterClient);
+    }
+  }
+
+  /**
+   * Deletes a lineage identified by a given id. If the delete is cascade, it will delete all the
+   * downstream lineages that depend on the given one recursively. Otherwise it does not delete the
+   * lineage exception, if there are other lineages whose input files are the output files of the
+   * specified lineage.
+   *
+   * @param lineageId the id of the lineage
+   * @param cascade whether to delete all the downstream lineages recursively
+   * @return true if the lineage deletion is successful, false otherwise
+   * @throws IOException if the master cannot delete the lineage
+   * @throws LineageDeletionException if the deletion is cascade but the lineage has children
+   * @throws LineageDoesNotExistException if the lineage does not exist
+   */
+  public boolean deleteLineage(long lineageId, boolean cascade)
+      throws IOException, LineageDoesNotExistException, LineageDeletionException {
+    LineageMasterClient masterClient = mContext.acquireMasterClient();
+    try {
+      boolean result = masterClient.deleteLineage(lineageId, cascade);
+      LOG.info(result ? "Succeeded to " : "Failed to" + "delete lineage " + lineageId);
+      return result;
+    } finally {
+      mContext.releaseMasterClient(masterClient);
+    }
+  }
+
+  /**
+   * Lists all the lineages.
+   *
+   * @return the informaiton about lineages
+   * @throws IOException if the master cannot list the lineage info
+   */
+  public List<LineageInfo> listLineages() throws IOException {
+    LineageMasterClient masterClient = mContext.acquireMasterClient();
+
+    try {
+      List<LineageInfo> result = masterClient.listLineages();
+      return result;
+    } finally {
+      mContext.releaseMasterClient(masterClient);
+    }
+  }
+
+  /**
+   * All the files are already created when lineage was added. This method reconfigures the created
+   * file.
+   *
+   * @throws IOException
+   */
+  public long recreate(TachyonURI path, long blockSize, boolean recursive, long ttl) {
+    LineageMasterClient masterClient = mContext.acquireMasterClient();
+    try {
+      long fileId = masterClient.recreateFile(path.getPath(), blockSize);
+      return fileId;
+    } catch (IOException e) {
+      throw new RuntimeException("recreation failed", e);
+    } finally {
+      mContext.releaseMasterClient(masterClient);
+    }
+  }
+
+  /**
+   * Gets the output stream for lineage job. If the file already exists on master, returns a dummpy
+   * output stream.
+   */
+  @Override
+  public FileOutStream getOutStream(TachyonURI path, OutStreamOptions options)
+      throws IOException, TachyonException {
+    long fileId = recreate(path, options.getBlockSize(), true, options.getTTL());
+    if (fileId < 0) {
+      return new DummyOutputStream(fileId, options);
+    }
+    return new LineageFileOutStream(fileId, options);
```

Comment:
Should check that this file is actually part of some lineage and not a normal file